src/decorationProvider.ts

```diff
@@ -45,16 +59,40 @@ export class JJDecorationProvider implements FileDecorationProvider {
     }
 
     this.decorations = nextDecorations;
-    this._onDidChangeDecorations.fire(
-      [...changedDecorationKeys.keys()].map((key) => {
+    this._onDidChangeDecorations.fire([
+      ...[...changedDecorationKeys.keys()].map((key) => {
         const [fsPath, rev] = key.split(":");
         return withRev(Uri.file(fsPath), rev);
       }),
-    );
+      ...[...changedDecorationKeys.keys()].map((key) => {
+        const [fsPath, rev] = key.split(":");
+        if (rev === '@') {
+          return Uri.file(fsPath);
```

Comment:
This must've been the fix for the bug you were talking about, right? I think I see the issue.